src/conn.rs

```diff
@@ -287,8 +306,24 @@ impl<const N: usize, P: ConnectionPeer> Connection<N, P> {
                 }
                 () = &mut idle_timeout => {
                     if !std::matches!(self.state, State::Closed { .. }) {
+                        // Warn that we are giving up on the connection due to a lack of activity.
+
+                        // First, we look for this special situation:
+                        // Our outgoing FIN was sent, but the incoming ACK was dropped.
+                        // This timeout is not worthy of a warning log, because everything
+                        // basically went fine: all data was sent and ACK'd.
+                        // There is no more work to do.
```

Comment:
I think this is a bit more clear/concise

```suggestion
                        // If both endpoints have exchanged FINs, and our FIN is our only unacknowledged
                        // packet, then do not emit a warning, since a single STATE packet could have simply
                        // been dropped
```

Reply:
Ok, I took another pass.